clang/lib/DPCT/Diagnostics/Diagnostics.inc

```diff
@@ -294,6 +294,8 @@ DEF_WARNING(UNSUPPORTED_KERNEL_ATTRIBUTE, 1132, HIGH_LEVEL, "SYCL 2020 does not
 DEF_COMMENT(UNSUPPORTED_KERNEL_ATTRIBUTE, 1132, HIGH_LEVEL, "SYCL 2020 does not support accessing the {0} for the kernel. The API is replaced with member variable \"{1}\". Please set the appropriate value for \"{1}\".")
 DEF_WARNING(SPARSE_NNZ, 1134, MEDIUM_LEVEL, "The tool cannot deduce the consumer API (\"dpct::sparse::csrgemm\") of this API, and this API has 2 arguments depending on the 8th and the 12th parameters of the consumer API. Please replace the 2 arguments tagged as \"dpct_placeholder\" with the corresponding value.")
 DEF_COMMENT(SPARSE_NNZ, 1134, MEDIUM_LEVEL, "The tool cannot deduce the consumer API (\"dpct::sparse::csrgemm\") of this API, and this API has 2 arguments depending on the 8th and the 12th parameters of the consumer API. Please replace the 2 arguments tagged as \"dpct_placeholder\" with the corresponding value.")
+DEF_WARNING(JOINT_MATRIX_SHAPE, 1135, HIGH_LEVEL, "Different hardware supports different combinations of the data type and the shape of the joint_matrix. Please check if the combination of the data type and the matrix shape of the below type meets the restrictions.")
```

Comment:
Please check if joint_matrix implementations support the combination of data type and matrix shape type in the target hardware.

Reply:
updated in [6c9db4f](https://github.com/oneapi-src/SYCLomatic/pull/2494/commits/6c9db4fd8fdc2c3007f08de8ee5ab1d7dcf578be)